pkg/printers/interface.go

```diff
@@ -72,9 +74,97 @@ type PrintOptions struct {
 	ColumnLabels       []string
 
 	SortBy string
+}
+
+// separate template flags
+type TemplatePrintFlags struct {
+	// indicates if it is OK to ignore missing keys for rendering
+	// an output template.
+	AllowMissingKeys *bool
+	TemplateArgument *string
+
+	// not actually bound to a flag - this may be set by
+	// a more general flags struct that composes this one,
+	// or its value may be provided via an arg to ToPrinter.
+	// We need this to select the appropriate PrintBuilder.
+	OutputFormat string
+
+	PrinterBuilders []func(*TemplatePrintFlags) (ResourcePrinter, bool, error)
+}
+
+func (f *TemplatePrintFlags) Validate() error {
+	if f.TemplateArgument == nil || len(*f.TemplateArgument) == 0 {
+		return fmt.Errorf("missing --template value")
+	}
+	return nil
+}
+
+func (f *TemplatePrintFlags) ToPrinter(outputFormat string) (ResourcePrinter, error) {
+	f.OutputFormat = outputFormat
+
```

Comment:
I am relaying the value of the `--output` flag by receiving it as an argument in this method.
I did not want to bind the --output flag to this struct, but have not yet thought of a cleaner way to obtain this value here.